### PR TITLE
Reorganize navigation menu with nav tabs

### PR DESCRIPTION
**Pull request by straight-shoota**
Implements proposal from https://github.com/crystal-lang/crystal-book/issues/384#issuecomment-799812240 based on mkdocs-material's nav-tabs feature.

Screenshot home page:
grafik

### CODE REVIEW

**Comment by straight-shoota** [comment]
Preview: https://straight-shoota.github.io/crystal-book/feature/nav-tabs

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-* [Syntax and semantics](syntax_and_semantics/README.md)
+* [Home](README.md)
+    * [Getting started](getting_started/README.md)
+        * [An HTTP Server](getting_started/http_server.md)
+        * [A Command Line Application](getting_started/cli.md)
+    * [Using the compiler](using_the_compiler/README.md)
+    * [The shards command](the_shards_command/README.md)
+    * [Coding style](conventions/coding_style.md)
+    * [Platform Support](platform_support.md)
+* [Specification](syntax_and_semantics/README.md)
```
Comment (oprypin) [inline docs/SUMMARY.md:11]:
```suggestion
* Specification
    * [Syntax and semantics](syntax_and_semantics/README.md)
```

File: docs/SUMMARY.md [hunk 2]
```diff
@@ -135,7 +131,10 @@
         * [Travis CI](guides/ci/travis.md)
         * [CircleCI](guides/ci/circleci.md)
     * [Static Linking](guides/static_linking.md)
-* [Platform Support](platform_support.md)
+    * [Database](database/README.md)
+        * [Connection](database/connection.md)
+        * [Connection pool](database/connection_pool.md)
+        * [Transactions](database/transactions.md)
 * [Tutorials](tutorials/README.md)
```
Comment (oprypin) [inline docs/SUMMARY.md:138]: I'd delete the index page for tutorials, it doesn't have any useful content
```suggestion
* Tutorials
```

Comment (straight-shoota) [reply docs/SUMMARY.md:138]: Not *yet*. There are already plans for follow-up courses... But yeah we can leave the tutorials landing page out as long es it has no meaningful content.

Comment (oprypin) [reply docs/SUMMARY.md:138]: *tutorials/basics/README.md* already looks like it wants to be the landing page, so let it!

File: docs/SUMMARY.md (hunk 1 above)
Comment (straight-shoota) [reply docs/SUMMARY.md:11]: Now *this* landing page also qualifies for "no useful content" :D

#480 doesn't really change that, either.

Comment (oprypin) [reply docs/SUMMARY.md]: If this PR lands first, then definitely delete that page in #480

**Review by oprypin — APPROVED** [review]

File: docs/SUMMARY.md (hunk 2 above)
Comment (oprypin) [reply docs/SUMMARY.md]: Oh still need to actually delete the page

> The following pages exist in the docs directory, but are not included in the "nav" configuration:
>  - tutorials/README.md

**Review by bcardiff — APPROVED** [review]

**Pull request merged**